custom/icds_reports/static/js/directives/info-message/info-message.directive.js

```diff
@@ -29,6 +32,18 @@ function InfoMessageController($location) {
             vm.currentMonth = moment().format("MMMM");
             vm.nextMonth = moment().startOf('month').add(1, 'months').format("MMMM");
             vm.year = moment().format('YYYY');
+            vm.yearOfPreviousMonth = vm.year;
+            vm.yearOfNextMonth = vm.year;
+            if (currentMonth === 1) {
+                vm.yearOfPreviousMonth = moment().startOf('year').subtract(1, 'years').format('YYYY');
```

Comment:
nit: I think this can just be `vm.yearOfPreviousMonth -= 1`, no?

Reply:
I think you also need to set `vm.yearOfTwoMonthsAgo` here (even if it's not used it just seems bad to leave the variable wrong).

Reply:
yearOfTwoMonthsAgo value will be based on yearOfPreviousMonth. So assigned it after yearOfPreviousMonth

Reply:
> nit: I think this can just be `vm.yearOfPreviousMonth -= 1`, no?

yearOfPreviousMonth is a string representation of the month. So cant do -=1

Reply:
Why are they strings?

Reply:
they are being directly accessed in the html. so intially they might have put it like that. I just add the new variables to it

Reply:
I believe it should all work with integers and then that would simplify things substantially though, no?

Reply:
yes. But anyways we are not reconverting them to int rather using the current month varibale which should be fine.

Reply:
@challabeehyv  As the Ticket looks, the priority if lower. We can use this opportunity to clean the code.
Here is some clue which you will give you some hint how you could do it here:
<img width="529" alt="Screenshot 2020-01-08 at 9 40 01 PM" src="https://user-images.githubusercontent.com/21310920/71994974-be0c9900-325f-11ea-8a4f-7d52747f674e.png">

Reply:
This single month subtract also moves back the Year beacuse you which you wont need to change the year manually.

Reply:
i was just following the earlier implementation. I will refactor it then.Thanks for  the suggest

Reply:
done